### ISSUE TITLE
clang-tidy check bugprone-use-after-move doesn't work with std::vector::emplace

### ISSUE THREAD

**Issue by sean-mcmanus** [llvm/llvm-project#55238]
I used clang-tidy 14.0.0 with something like `clang-tidy.exe --checks=bugprone-use-after-move <file> --`

With C++ code
```cpp
#include <unordered_map>
#include <string>

int f(std::string &&s) { return s.length(); }
int main()
{
    std::string s = "1";
    std::unordered_map<std::string, int> v;
    v.emplace(s, f(std::move(s)));
    return 0;
}
```

Optionally you can add code below the repro code like
```cpp
#include <vector>
void g(int i, std::vector<int> v);
int main2()
{
std::vector<int> v = { 1, 2, 3 };
g(v[1], std::move(v));
return 0;
}
```
from https://releases.llvm.org/14.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/bugprone-use-after-move.html#unsequenced-moves-uses-and-reinitializations to verify the expected behavior.

Bug: No clang-tidy warning is reported, even though at runtime the moved string is used (at least with some compilers).

If this can't be fixed for some "by design" reason, then I would appreciate it if the documentation would describe the conditions which prevent the check from working.

**Comment by PiotrZSL** [comment]
Problem exists because check is trying to be too smart, and only detect issues when std::move would actually result in direct call to move constructor, but don't detect passing as rvalue or by using perfect forwarding.

Example:

```c++
void testUnc(std::string&&, const char&);

void test(std::string p)
{
    testUnc(std::move(p), p[0]);
}
```

We already did a move, but it didn't result in call to move constructor. If check would be more dumb, then these issues would be detected.

**Comment by vbvictor** [comment]
https://godbolt.org/z/sjePT4ra9
